airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSource.java

```diff
@@ -115,9 +115,15 @@ public Set<JdbcPrivilegeDto> getPrivilegesTableForCurrentUser(final JdbcDatabase
         connection -> {
           connection.setAutoCommit(true);
           final PreparedStatement ps = connection.prepareStatement(
-              "SELECT schemaname, tablename "
-                  + "FROM   pg_tables "
-                  + "WHERE  has_table_privilege(schemaname||'.'||tablename, 'select') = true AND schemaname = ?;");
+              "select schemaname, tablename "
+                + "from (SELECT "
+                + "  v.schemaname as schemaname, "
+                + "  v.resourcename as tablename "
+                + "FROM "
+                + "  (SELECT schemaname, viewname as resourcename FROM pg_views "
+                + "  UNION ALL SELECT schemaname, tablename as resourcename from pg_tables) v "
+                + "WHERE "
+                + "    has_table_privilege(current_user, v.schemaname || '.' || quote_ident(v.resourcename), 'select') = true and v.schemaname = ?) as permission;");
```

Comment:
do you need all of the `as ....` in there given you're not really renaming things?  seems like maybe only the `viewname as tablename` may strictly be needed and might make readability a little better.  It also seems like the outer select could be removed (not sure-- with the formatting here it's hard to see-- it looks to me like this is doing `select schemaname, tablename from (...) as permission;` which I don't think is valid?